### DISCUSSION TITLE
Is S3 supported?

### DISCUSSION THREAD

**Discussion by ghost** [gofr-dev/gofr#1200] [Q&A]
As per the code I can see there is a s3 directory inside `datasoruce`. This API is not documented however, is this API supported and stbale?

**Comment by coolwednesday** [comment]
Yes, working with s3 is similar as working with other file-stores that you can find here : https://gofr.dev/docs/advanced-guide/handling-file

You can add s3 filestore as follows :
`go get gofr.dev/pkg/gofr/datasource/file/s3@latest`

```
import "gofr.dev/pkg/gofr/datasource/file/s3"

func main() {
	app := gofr.New()

	app.AddFileStore(s3.New(&s3.Config{
		EndPoint:        "http://localhost:4566",
		BucketName:      "gofr-bucket-2",
		Region:          "us-east-1",
		AccessKeyID:     "test",
		SecretAccessKey: "test",
	}))

	// Run the app
	app.Run()
}
```

Please note that the current implementation allows you to handle only one bucket at one time, gofr-bucket-2 in this case. You cannot change the bucket in between. In S3, all the paths must be full path relative to the s3 bucket. 

Thanks for pointing out ! We will update the documentation as soon as possible.
